src/commands/signzone.rs

```diff
@@ -0,0 +1,285 @@
+    #[arg(short = 'n', default_value_t = false, group = "nsec3")]
+    use_nsec3: bool,
+
+    /// Hashing algorithm
+    #[arg(
+        help_heading = Some("NSEC3 (when using '-n')"),
+        short = 'a',
+        value_name = "algorithm",
+        default_value = "SHA-1",
+        value_parser = ValueParser::new(Nsec3Hash::parse_nsec_alg),
+        requires = "nsec3"
+    )]
+    algorithm: Nsec3HashAlg,
+
+    /// Number of hash iterations
+    // TODO: make the default for dnst-signzone be 0 (to match best practice)
+    // while leaving the default for ldns-signzone be 1 (to match ldns), or
+    // maybe even change the default for both to 0.
+    #[arg(
+        help_heading = Some("NSEC3 (when using '-n')"),
+        short = 't',
+        value_name = "number",
+        default_value_t = 1,
+        requires = "nsec3"
+    )]
+    iterations: u16,
+
+    /// Salt
+    #[arg(
+        help_heading = Some("NSEC3 (when using '-n')"),
+        short = 's',
+        value_name = "string",
+        default_value_t = Nsec3Salt::empty(),
+        requires = "nsec3"
+    )]
+    salt: Nsec3Salt<Bytes>,
+
+    /// Set the opt-out flag on all NSEC3 RRs
+    #[arg(
+        help_heading = Some("NSEC3 (when using '-n')"),
+        short = 'p',
+        default_value_t = false,
+        requires = "nsec3",
+        conflicts_with = "nsec3_opt_out"
+    )]
+    nsec3_opt_out_flags_only: bool,
+
+    /// Set the opt-out flag on all NSEC3 RRs and skip unsigned delegations
+    #[arg(
+        help_heading = Some("NSEC3 (when using '-n')"),
+        short = 'A', // Matches BIND dnssec-signzone
+        default_value_t = false,
+        requires = "nsec3",
+        conflicts_with = "nsec3_opt_out_flags_only"
+    )]
+    nsec3_opt_out: bool,
+
+    /// Hash only, don't sign
+    #[arg(short = 'H', default_value_t = false)]
+    hash_only: bool,
+
+    /// Use layout in signed zone and print comments on DNSSEC records
+    #[arg(short = 'b', default_value_t = false)]
+    diagnostic_comments: bool,
+
+    /// Origin for the zone (for zonefiles with relative names and no $ORIGIN)
+    #[arg(short = 'o')]
+    origin: Option<Name<Bytes>>,
+
+    /// The zonefile to sign
+    #[arg(value_name = "zonefile")]
+    zonefile_path: PathBuf,
+
+    /// The keys to sign the zone with
+    ///
+    /// keys must be specified by their base name (usually
+    /// K<name>+<alg>+<id>), i.e. WITHOUT the .private extension.
+    #[arg(value_name = "key")]
+    key_paths: Vec<PathBuf>,
+}
+
+impl SignZone {
+    pub fn execute(self) -> Result<(), Error> {
+        let mut records = self.load_zone()?;
+
+        //---
+
+        // Import the specified key.
+        let mut keys = vec![];
+        for key_path in self.key_paths {
+            let old_ext = key_path.extension().unwrap().to_str().unwrap();
+            let new_ext = format!("{}.private", old_ext);
+            let private_key_path = key_path.with_extension(new_ext).display().to_string();
```

Comment:
I really dislike using `.display().to_string()` as the path you pass into fs operations.  Once you've appended the extension, use the `PathBuf` as is, and when you need to print it for errors, just call `.display()` there.